### PR TITLE
Update gradle and build.gradle

### PR DESCRIPTION
**Pull request by lbergelson**
* update gradle wrapper 8.2.1 ->  8.10.2
* remove 'versions' plugin because we don't use it
* update gradle plugins to new versions
  * shadow plugin changed maintainers and coordinates com.github.johnrengelman.shadow:8.1.1 -> com.gradleup.shadow:8.3.3
  * git-version 0.5.1 -> 3.1.0
  * sonatype scan 2.6.1 -> 2.8.3
  * download 5.4.0 -> 5.6.0
* use tasks.register() which is the newer style

### CODE REVIEW

**Comment by lbergelson** [comment]
@droazen @KevinCLydon If one of you wants to take a look at this that would be great. It updates a bunch of build and housekeeping stuff.  It doesn't resolve all the dependency vulnerabilities yet.

File: gradlew [hunk 1]
```diff
@@ -80,13 +80,10 @@ do
     esac
 done
 
-APP_HOME=$( cd "${APP_HOME:-./}" && pwd -P ) || exit
-
-APP_NAME="Gradle"
```
Comment (jamesemery) [inline gradlew:85]: does it not need to specify appname anymore?

**Review by jamesemery — APPROVED** [review]
nothing objectionable here

File: gradlew (hunk 1 above)
Comment (lbergelson) [reply gradlew:85]: This is autogenerated.  I just roll with whatever they suggest.

**Pull request merged**